File: chambers/src/main/java/vip/qoriginal/quantumplugin/chambers/ChamberRegion.java
package vip.qoriginal.quantumplugin.chambers;

import org.bukkit.Location;
import org.bukkit.World;

public record ChamberRegion(
        int minX,
        int minY,
        int minZ,
        int maxX,
        int maxY,
        int maxZ
) {
    public boolean contains(Location location, World instanceWorld, Location origin) {
        int offsetX = origin.getBlockX();
        int offsetY = origin.getBlockY();
        int offsetZ = origin.getBlockZ();
        return location.getWorld() != null
                && location.getWorld().getUID().equals(instanceWorld.getUID())
                && location.getBlockX() >= offsetX + minX
                && location.getBlockX() <= offsetX + maxX
                && location.getBlockY() >= offsetY + minY
                && location.getBlockY() <= offsetY + maxY
                && location.getBlockZ() >= offsetZ + minZ
                && location.getBlockZ() <= offsetZ + maxZ;
    }
}

File: chambers/src/main/java/vip/qoriginal/quantumplugin/chambers/data/ChamberRegion.kt
package vip.qoriginal.quantumplugin.chambers.data

import org.bukkit.Location
import org.bukkit.World

data class ChamberRegion(
    val minX: Int,
    val minY: Int,
    val minZ: Int,
    val maxX: Int,
    val maxY: Int,
    val maxZ: Int,
) {
    fun contains(location: Location, instanceWorld: World, origin: Location): Boolean {
        val offsetX = origin.blockX
        val offsetY = origin.blockY
        val offsetZ = origin.blockZ
        return location.world.uid == instanceWorld.uid &&
            location.blockX in (offsetX + minX)..(offsetX + maxX) &&
            location.blockY in (offsetY + minY)..(offsetY + maxY) &&
            location.blockZ in (offsetZ + minZ)..(offsetZ + maxZ)
    }
}